src/main/webapp/app/exercises/shared/dashboards/tutor/exercise-assessment-dashboard.component.ts

```diff
@@ -598,47 +598,63 @@ export class ExerciseAssessmentDashboardComponent implements OnInit {
         let participationId = undefined;
         if (submission !== 'new' && submission.participation !== undefined) {
             participationId = submission.participation!.id;
         }
-        const url = getLinkToSubmissionAssessment(this.exercise.type!, this.courseId, this.exerciseId, participationId, submissionId, this.examId, this.exerciseGroupId);
+        return getLinkToSubmissionAssessment(this.exercise.type!, this.courseId!, this.exerciseId, participationId, submissionUrlParameter, this.examId, this.exerciseGroupId);
+    }
+
+    /**
+     * Generates and returns the query parameters required for opening the assessment editor
+     * @param correctionRound
+     */
+    getAssessmentQueryParams(correctionRound = 0): object {
         if (this.isTestRun) {
-            this.router.navigate(url, { queryParams: { testRun: this.isTestRun, 'correction-round': correctionRound } });
+            return {
+                testRun: this.isTestRun,
+                'correction-round': correctionRound,
+            };
         } else {
-            this.router.navigate(url, { queryParams: { 'correction-round': correctionRound } });
+            return { 'correction-round': correctionRound };
         }
-        this.openingAssessmentEditorForNewSubmission = false;
     }
 
     /**
-     * Show complaint depending on the exercise type
-     * @param complaint that we want to show
+     * Generates and returns the query parameters required for opening a complaint
+     * @param complaint
      */
-    viewComplaint(complaint: Complaint) {
+    getComplaintQueryParams(complaint: Complaint) {
         const submission: Submission = complaint.result?.submission!;
         // numberOfAssessmentsOfCorrectionRounds size is the number of correction rounds
         if (complaint.complaintType === ComplaintType.MORE_FEEDBACK) {
-            this.openAssessmentEditor(submission, this.numberOfAssessmentsOfCorrectionRounds.length - 1);
+            return this.getAssessmentQueryParams(this.numberOfAssessmentsOfCorrectionRounds.length - 1);
+        }
+        const result = this.getSubmissionToViewFromComplaintSubmission(submission);
+
+        if (result !== undefined) {
+            return this.getAssessmentQueryParams(result.results!.length - 1);
         }
+    }
+
+    /**
+     * Returns either the corresponding submission to view or undefined.
+     * This complaint has to be a true complaint or else issues can arise.
+     * @param submission
+     */
+    getSubmissionToViewFromComplaintSubmission(submission: Submission) {
         const submissionToView = this.submissionsWithComplaints.filter((dto) => dto.submission.id === submission.id).pop()?.submission;
         if (submissionToView) {
             if (!submissionToView.results) {
                 submissionToView.results = [];
             }
             submissionToView.results = submissionToView.results?.filter((result) => result.assessmentType !== AssessmentType.AUTOMATIC);
```

Comment:
This line and the if above could be shortened to 
```suggestion
            submissionToView.results = submissionToView.results?.filter((result) => result.assessmentType !== AssessmentType.AUTOMATIC) ?? [];
```
however I am not sure if it improves readability

Reply:
I guess this is ok. Otherwise, we can remove the `?` and put the `filter` in an `else` block.

Reply:
I decided to leave it as a standard if clause but I changed it to an if-else clause as suggested because I think this is more intuitive to people not as familiar with TS syntax.